### PR TITLE
fix(fuentbit): update PSP resource with missing values

### PR DESCRIPTION
**Pull request by aslafy-z**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes https://github.com/banzaicloud/logging-operator/issues/731
| License         | Apache 2.0

### What's in this PR?

Update the Fluentbit PSP with missing hostpaths, hostports and hostnetwork.

### Why?

Logging operator cannot be deployed on GKE with the current default.

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)

### CODE REVIEW

File: pkg/resources/fluentbit/psp.go [hunk 1]
```diff
@@ -53,14 +60,32 @@ func (r *Reconciler) clusterPodSecurityPolicy() (runtime.Object, reconciler.Desi
...
+		hostPorts := []policyv1beta1.HostPortRange{}
+		if r.Logging.Spec.FluentbitSpec.HostNetwork {
+			if r.Logging.Spec.FluentbitSpec.Metrics != nil && r.Logging.Spec.FluentbitSpec.Metrics.Port != 0 {
+				hostPorts = append(hostPorts, policyv1beta1.HostPortRange{
+					Min: r.Logging.Spec.FluentbitSpec.Metrics.Port,
+					Max: r.Logging.Spec.FluentbitSpec.Metrics.Port,
+				})
+			}
+
+			if r.Logging.Spec.FluentbitSpec.Metrics != nil && r.Logging.Spec.FluentbitSpec.BufferVolumeMetrics.Port != 0 {
```
Comment (siliconbrain) [inline pkg/resources/fluentbit/psp.go:72]:
```suggestion
			if r.Logging.Spec.FluentbitSpec.BufferVolumeMetrics != nil && r.Logging.Spec.FluentbitSpec.BufferVolumeMetrics.Port != 0 {
```

**Review by siliconbrain — CHANGES_REQUESTED** [review]

File: pkg/resources/fluentbit/psp.go [hunk 2]
```diff
@@ -36,6 +36,13 @@ func (r *Reconciler) clusterPodSecurityPolicy() (runtime.Object, reconciler.Desi
 			ReadOnly:   true,
 		}}
 
+		if r.Logging.Spec.FluentbitSpec.BufferStorageVolume.HostPath != nil {
```
Comment (siliconbrain) [inline pkg/resources/fluentbit/psp.go:39]:
```suggestion
		if r.Logging.Spec.FluentbitSpec.BufferStorageVolume != nil && r.Logging.Spec.FluentbitSpec.BufferStorageVolume.HostPath != nil {
```

**Review by siliconbrain — CHANGES_REQUESTED** [review]

**Review by siliconbrain — APPROVED** [review]

File: pkg/resources/fluentbit/psp.go [hunk 3]
```diff
@@ -36,6 +36,13 @@ func (r *Reconciler) clusterPodSecurityPolicy() (runtime.Object, reconciler.Desi
 			ReadOnly:   true,
 		}}
 
+		if r.Logging.Spec.FluentbitSpec.BufferStorageVolume != nil && r.Logging.Spec.FluentbitSpec.BufferStorageVolume.HostPath != nil {
```
[the remark this answers is not in the document]
Comment (aslafy-z) [reply pkg/resources/fluentbit/psp.go:39]: I guess I also misread the type, I checked your proposals I guess wrongly :)

**Review by siliconbrain — APPROVED** [review]

**Review by ahma — APPROVED** [review]
LGTM Thanks

**Pull request merged**